contrib/oauth2/mutt_oauth2.py

```diff
@@ -92,8 +93,26 @@
 ap.add_argument('-a', '--authorize', action='store_true', help='manually authorize new tokens')
 ap.add_argument('--authflow', help='authcode | localhostauthcode | devicecode')
 ap.add_argument('-t', '--test', action='store_true', help='test IMAP/POP/SMTP endpoints')
+ap.add_argument('--decryption-pipe', type=shlex.split, default=DECRYPTION_PIPE,
+        help='decryption command (string) default: "{}"'.format(
+            " ".join(DECRYPTION_PIPE)))
+ap.add_argument('--encryption-pipe', type=shlex.split, default=ENCRYPTION_PIPE,
+        help='encryption command (string) default: "{}"'.format(
+            " ".join(ENCRYPTION_PIPE)))
+ap.add_argument('--google-id', type=str, default='', help='')
+ap.add_argument('--google-secret', type=str, default='', help='')
+ap.add_argument('--microsoft-id', type=str, default='', help='')
+ap.add_argument('--microsoft-secret', type=str, default='', help='')
 args = ap.parse_args()
 
+ENCRYPTION_PIPE = args.encryption_pipe
+DECRYPTION_PIPE = args.decryption_pipe
+
+registrations['google']['client_id'] = args.google_id
+registrations['google']['client_secret'] = args.google_secret
+registrations['microsoft']['client_id'] = args.microsoft_id
+registrations['microsoft']['client_secret'] = args.microsoft_secret
```

Comment:
Why do we need split flags? Client_id / client_secret should do no? Then just add a provider switch to pre populate the endpoints.

Also introducing options without a help seems somewhat bad form.